### PR TITLE
fix(event search) Correctly handle trailing quotes

### PR DESCRIPTION
**Pull request by evanh**
If a search query has trailing quotes, then require that they be escaped or to
add another quote to avoid the trailing quotes.

### CODE REVIEW

File: src/sentry/api/event_search.py [hunk 1]
```diff
@@ -558,7 +557,18 @@ def visit_boolean_operator(self, node, children):
         return node.text
 
     def visit_value(self, node, children):
-        return node.text
+        # A properly quoted value will match the quoted value regex, so any unescaped
+        # quotes are errors.
+        for (idx, c) in enumerate(node.text):
```
Comment (markstory) [inline src/sentry/api/event_search.py]: Do you need to enumerate all the characters for this check? It seems like something you could do with a regular expression.

**Review by markstory — APPROVED** [review]

File: tests/sentry/api/test_event_search.py [hunk 2]
```diff
@@ -414,11 +412,50 @@ def test_empty_filter_value(self):
...
+            SearchFilter(
+                key=SearchKey(name="device.family"),
+                operator="=",
+                value=SearchValue(raw_value='te"st'),
+            )
+        ]
+
+        # This is a weird case. I think this should be an error, but it doesn't seem trivial to rewrite
+        # the grammar to handle that.
+        assert parse_search_query('url:"te"st') == [
```
Comment (dashed) [inline tests/sentry/api/test_event_search.py]: Definitely not what I expected.

**Review by dashed — APPROVED** [review]
Looks good to me 👍

File: src/sentry/api/event_search.py (hunk 1 above)
Comment (evanh) [reply src/sentry/api/event_search.py]: I don't know that a regex would be more performant than enumerate but I can check.

Comment (evanh) [reply src/sentry/api/event_search.py]: Well normally I would say if you solve a problem with regex then you have two problems. After testing though, the regex option is noticeably faster than the non-regex version. Thanks Python.

Comment (markstory) [reply src/sentry/api/event_search.py]: Probably significantly fewer allocations being made.

**Review by markstory — APPROVED** [review]

**Review by markstory — APPROVED** [review]

**Review by dashed — APPROVED** [review]
Looks good to me again 👍

**Review by markstory — APPROVED** [review]

**Pull request merged**